[CI] ~/.docker/config.json issues (file not found) - mandrel docker client header step

## Description

After #641 we still see issues in CI related to setting the user agent in docker config:

```
 jq: error: Could not open file /home/runner/.docker/config.json: No such file or directory
/home/runner/work/_temp/61a2380e-b2d9-4100-b3a7-1686a4b14df3.sh: line 1: /home/runner/.docker/config.json.new: No such file or directory
Error: Process completed with exit code 1.
```

See for example:
https://github.com/graalvm/mandrel/actions/runs/7248996212/job/19747102268?pr=640#step:10:25

Looks to me that those expected files `~/.docker/config.json` aren't there on some runners to begin with. We need to figure out what creates them and run those commands first.

--- a/.github/workflows/base-windows.yml
+++ b/.github/workflows/base-windows.yml
@@ -468,8 +468,10 @@ jobs:
         if: startsWith(matrix.os-name, 'windows')
         shell: bash
         run: |
-          cat <<< $(jq '.HttpHeaders += {"User-Agent": "Mandrel-CI-Docker-Client"}' ~/.docker/config.json) > ~/.docker/config.json.new
-          mv ~/.docker/config.json.new ~/.docker/config.json
+          if [ -f ~/.docker/config.json ]; then
+            cat <<< $(jq '.HttpHeaders += {"User-Agent": "Mandrel-CI-Docker-Client"}' ~/.docker/config.json) > ~/.docker/config.json.new
+            mv ~/.docker/config.json.new ~/.docker/config.json
+          fi
       - name: Change quarkus.version for Quarkus 2.2 to make mandrel-integration-test not apply quarkus_main.patch
         if: startsWith(matrix.os-name, 'windows')
         # See https://github.com/Karm/mandrel-integration-tests/pull/64
@@ -611,8 +613,10 @@ jobs:
       - name: Update Docker Client User Agent
         shell: bash
         run: |
-          cat <<< $(jq '.HttpHeaders += {"User-Agent": "Mandrel-CI-Docker-Client"}' ~/.docker/config.json) > ~/.docker/config.json.new
-          mv ~/.docker/config.json.new ~/.docker/config.json
+          if [ -f ~/.docker/config.json ]; then
+            cat <<< $(jq '.HttpHeaders += {"User-Agent": "Mandrel-CI-Docker-Client"}' ~/.docker/config.json) > ~/.docker/config.json.new
+            mv ~/.docker/config.json.new ~/.docker/config.json
+          fi
       - name: Build with Maven
         run: |
           cmd.exe /c "call `"C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat`" && set > %temp%\vcvars64.txt"

--- a/.github/workflows/base.yml
+++ b/.github/workflows/base.yml
@@ -542,8 +542,10 @@ jobs:
         if: "!startsWith(matrix.os-name, 'windows')"
         shell: bash
         run: |
-          cat <<< $(jq '.HttpHeaders += {"User-Agent": "Mandrel-CI-Docker-Client"}' ~/.docker/config.json) > ~/.docker/config.json.new
-          mv ~/.docker/config.json.new ~/.docker/config.json
+          if [ -f ~/.docker/config.json ]; then
+            cat <<< $(jq '.HttpHeaders += {"User-Agent": "Mandrel-CI-Docker-Client"}' ~/.docker/config.json) > ~/.docker/config.json.new
+            mv ~/.docker/config.json.new ~/.docker/config.json
+          fi
       - name: Change quarkus.version for Quarkus 2.2 to make mandrel-integration-test not apply quarkus_main.patch
         if: "!startsWith(matrix.os-name, 'windows')"
         # See https://github.com/Karm/mandrel-integration-tests/pull/64
@@ -697,8 +699,10 @@ jobs:
           tar -xzvf jdk.tgz -C ${GRAALVM_HOME} --strip-components=1
       - name: Update Docker Client User Agent
         run: |
-          cat <<< $(jq '.HttpHeaders += {"User-Agent": "Mandrel-CI-Docker-Client"}' ~/.docker/config.json) > ~/.docker/config.json.new
-          mv ~/.docker/config.json.new ~/.docker/config.json
+          if [ -f ~/.docker/config.json ]; then
+            cat <<< $(jq '.HttpHeaders += {"User-Agent": "Mandrel-CI-Docker-Client"}' ~/.docker/config.json) > ~/.docker/config.json.new
+            mv ~/.docker/config.json.new ~/.docker/config.json
+          fi
       - name: Install gdb
         run: |
           sudo apt-get update -y

